ksql-cli/src/main/java/io/confluent/ksql/util/TabularRow.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.rest.entity.FieldInfo;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class TabularRow {
+
+  private final int width;
+  private final List<String> value;
+  private final List<String> header;
+
+  public TabularRow(
+      final int width,
+      final List<FieldInfo> header,
+      final GenericRow value
+  ) {
+    this(
+        width,
+        header.stream().map(FieldInfo::getName).collect(Collectors.toList()),
+        value == null
+            ? null
+            : value.getColumns().stream().map(Objects::toString).collect(Collectors.toList()));
+  }
+
+  @VisibleForTesting
+  TabularRow(
+      final int width,
+      final List<String> header,
+      final List<String> value
+  ) {
+    this.header = Objects.requireNonNull(header, "header");
+    this.width = width;
+    this.value = value;
+  }
+
+  @Override
+  public String toString() {
```

Comment:
this method would benefit from being split into several methods, rather than being long with comments.

Something like:

```java
public String toString() {
final List<String> columns = value == null ? header : value;

     final int cellWidth = width / columns.size() - 2;
    final String format = "%-" + cellWidth + "s|";

    final List<List<String>> split = fitToCellWidth();

    final int maxSplit = maxLines(split);

    return formatRow(split, cellWidth, ...)
       + buildDiv(coluns.size(), cellWidth);
}
```